alcs-frontend/src/app/features/application/documents/document-upload-dialog/document-upload-dialog.component.ts

```diff
@@ -98,7 +110,10 @@ export class DocumentUploadDialogComponent implements OnInit, OnDestroy {
       visibilityFlags,
     };
     this.isSaving = true;
-    if (this.data.existingDocument) {
+    if (parcelId && dto.file && this.type.value === DOCUMENT_TYPE.CERTIFICATE_OF_TITLE) {
+      // @ts-ignore File is checked above to not be undefined, typescript still thinks it might be undefined
+      await this.applicationDocumentService.attachCertificateOfTitle(this.data.fileId, parcelId, dto);
```

Comment:
Maybe this one will be better than ts-ignore 
this.data.fileId! 
or just check in if statement

Reply:
Cleaned up!